File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    /**
    imput text
    calculate the letters ( excluding space) - loop
    count spaces to count words - loop
        - spaces count +1 = word count
    count sentences by counting . ! ? - loop
    calculate the Coleman-Lyau index
        index = 0.0588 * L - 0.296 * S - 15.8
        output: rounded up whole number
    **/

    string text = get_string("Text: ");

    //calculating L
    int letters = 0; //total letters

    for (int i = 0; i < strlen(text); i++) // counting letters propely
    {
        if (isalpha(text[i])) //checking for letters
        {
            letters++;
        }
    }

    int spaces = 0; //spaces count
    int sentences = 0; // total sencences

    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i])) //checking for spaces
        {
            spaces++;
        }
        else if (text[i] == 46) //checking for .
        {
            sentences++;
        }
        else if (text[i] == 33) //checking for !
        {
            sentences++;
        }
        else if (text[i] == 63) //checking for ?
        {
            sentences++;
        }
    }
    
    int words = spaces + 1; // counting the words

    float L = (float) letters / words * 100; //l index
    float S = (float) sentences / words * 100; //S index

    //printf("letters:%i words:%i sentences=%i\n", letters, words, sentences);

    int index = (float) round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
import cs50
import re

# get text
text = cs50.get_string("Text: ")

# count letters
letters = 0
for i in text:
    if(i.isalpha()):
        letters += 1

# count words
words = 0
word_list = text.split()
words = len(word_list)

# count sentences
sentences = len(re.split(r'[.!?]+', text)) - 1

# calculating the index

# L = average number of letters per 100 words in the text
L = (letters / words) * 100

# S = average number of sentences per 100 words in the text
S = (sentences / words) * 100

Coleman_Liau_index = round(0.0588 * L - 0.296 * S - 15.8)

# printing the results
if Coleman_Liau_index < 1:
    print("Before Grade 1")
elif Coleman_Liau_index > 16:
    print("Grade 16+")
else:
    print(f"Grade {Coleman_Liau_index}")

# debug
#print("Letters: ", letters)
#print("words: ", words)
#print("sentences: ", sentences)
#print("L: ", L)
#print("S: ", S)
#print("Coleman_Liau_index: ", Coleman_Liau_index)